src/__tests__/connection/passwordless/passwordless.actions.test.js

```diff
@@ -0,0 +1,231 @@
+import passwordless from 'connection/passwordless/actions';
+import { expectMockToMatch } from 'testUtils';
+
+jest.useFakeTimers();
+
+describe('passwordless actions', () => {
+  let mockFns;
+  let actions;
+  beforeEach(() => {
+    jest.resetModules();
+
+    jest.mock('connection/passwordless/index', () => ({
+      isEmail: jest.fn(),
+      isSendLink: jest.fn(),
+      resend: 'resend',
+      restartPasswordless: jest.fn(),
+      send: () => 'send',
+      setPasswordlessStarted: jest.fn(),
+      setResendFailed: jest.fn(),
+      setResendSuccess: jest.fn()
+    }));
+    jest.mock('field/phone_number', () => ({
+      phoneNumberWithDiallingCode: () => 'phoneNumberWithDiallingCode'
+    }));
+    jest.mock('field/index', () => ({
+      getFieldValue: (m, field) => field
+    }));
+    jest.mock('core/web_api', () => ({
+      startPasswordless: jest.fn(),
+      passwordlessVerify: jest.fn()
+    }));
+    jest.mock('core/actions', () => ({
+      closeLock: jest.fn(),
+      logIn: jest.fn(),
+      validateAndSubmit: jest.fn()
+    }));
+    jest.mock('i18n', () => ({ html: (_, keys) => keys.join(',') }));
+    jest.mock('core/index', () => ({
+      id: () => 'id',
+      setSubmitting: jest.fn(m => m),
+      auth: {
+        params: () => ({
+          toJS: () => ({
+            auth: 'params'
+          })
+        })
+      }
+    }));
+    jest.mock('store/index', () => ({
+      read: jest.fn(() => 'model'),
+      getEntity: 'getEntity',
+      swap: jest.fn(),
+      updateEntity: 'updateEntity'
+    }));
+
+    actions = require('connection/passwordless/actions');
+  });
+  describe('requestPasswordlessEmail()', () => {
+    it('calls validateAndSubmit()', () => {
+      actions.requestPasswordlessEmail('id');
+      expectMockToMatch(require('core/actions').validateAndSubmit, 1);
+    });
+    it('calls startPasswordless', () => {
+      actions.requestPasswordlessEmail('id');
```

Comment:
I see this one is not being validated. Is the widget validating the emails at least?

Reply:
yes, it uses the same component as lock, so has the same validations
![image](https://user-images.githubusercontent.com/941075/35593981-cfd1d604-05f8-11e8-887f-4a0f84b90494.png)